nixos/modules/services/networking/3proxy.nix

```diff
@@ -0,0 +1,427 @@
+{ config, lib, pkgs, ... }:
+with lib;
+let
+  pkg = pkgs._3proxy;
+  cfg = config.services._3proxy;
+  optionalList = list:
+    if list == [ ] then "*" else concatMapStringsSep "," toString list;
+  configFile = if cfg.confFile != null then
+    cfg.confFile
+  else
+    (pkgs.writeText "3proxy.conf" ''
```

Comment:
```diff
-  configFile = if cfg.confFile != null then
-    cfg.confFile
-  else
-    (pkgs.writeText "3proxy.conf" ''
+ services._3proxy.confFile = lib.mkDefaultOption (pkgs.writeText "3proxy.conf" ''
```
and move it downstairs from the `let`-block to under `config = mkIf cfg.enable {`

Reply:
I'm getting ```error: attribute 'configFile' missing``` when it's like that.

Reply:
https://github.com/NixOS/nixpkgs/pull/67507#discussion_r334259973

Reply:
> #67507 (comment)

Yes, i did that, but it's failing anyway.

Reply:
Figured this out. Done.